assembly/NFT/NFTWrapper.ts

```diff
@@ -0,0 +1,121 @@
+import {Address, call} from '../std';
+import {ByteArray} from '@massalabs/as/assembly';
+
+/**
+ * The Massa's standard NFT implementation wrapper.
+ *
+ * This class can be used to wrap a smart contract implementing
+ * Massa standard NFT.
+ * All the serialization/deserialization will handled here.
+ *
+ * ```assemblyscript
+ *  ...
+ *  const coin = new NFTWrapper(sc_address);
+ *  const coinName = coin.Name();
+ *  ...
+ * ```
+ */
+export class NFTWrapper {
+  _origin: Address;
+  _name: string;
+  _symbol: string;
+  _baseURI: string;
+  _limitSupply: string;
+
+  /**
+   * Wraps a smart contract exposing standard token FFI.
+   *
+   * @param {Address} at - Address of the smart contract.
+   */
+  constructor(at: Address) {
+    this._origin = at;
+    this._name = call(this._origin, 'name', '', 0);
+    this._symbol = call(this._origin, 'symbol', '', 0);
+    this._baseURI = call(this._origin, 'baseURI', '', 0);
+    this._limitSupply = call(this._origin, 'limitSupply', '', 0);
+    call(this._origin, 'setNFT', '', 0);
+  }
+
+  /**
+   * Return the NFT's name
+   * @return {string}
+   */
+  name(): string {
+    return this._name;
+  }
+  /**
+   * Return the NFT's symbol
+   * @return {string}
+   */
+  symbol(): string {
+    return this._symbol;
+  }
+
+  /**
+   * Return the token URI (external link written in NFT where pictures or others are stored)
+   * @param {u64} tokenId
+   * @return {string}
+   */
+  tokenURI(tokenId: u64): string {
+    return this._baseURI + tokenId.toString();
+  }
+
+  /**
+   * Return the base URI (external link written in NFT where pictures or others a stored)
+   * @return {string}
+   */
+  baseURI(): string {
+    return call(this._origin, 'baseURI', '', 0);
```

Comment:
As I have suggested here https://github.com/massalabs/massa-as-sdk/pull/28#discussion_r1016769927
tokenURI will return the old _baseURI if we do not update here.